index.js

```diff
@@ -63,6 +81,31 @@ function handleError( error, id, url ) {
 	return bot.sendMessage( id, error.message );
 }
 
+function sendUnfollowAcknowledgement( id, blog, count ) {
+	if ( count === 0) {
+		bot.sendMessage( id, `It seems you were not following ${blog}` );
+	} else {
+		bot.sendMessage( id, `No longer following ${blog}` );
+	}
+}
+
+function processCommand( id, command ) {
+	if ( command.method === 'follow' ) {
+		// we do not send a bot response yet:
+		// the response to xmpp sub command will trigger the response
+		return Promise.resolve()
+			.then( () => xmpp.subscribe( blogPath( command.blog ), id ) );
+	}
+	if ( command.method === 'unfollow' ) {
+		// we do not send an xmpp unsub command yet:
+		// other channels may have a subscription to this same blog.
+		return Promise.resolve()
+			.then( () => db.unfollowBlog( id, blogPath( command.blog ) ) )
+			.then( ( result ) => sendUnfollowAcknowledgement( id, command.blog, result ) );
+	}
+	return Promise.resolve();
```

Comment:
My understanding is that we are failing here, why not reject?

Reply:
true -if extractCommand didn't receive a command then the two handlers will currently abort without calling this function so i only added that final return because tern was complaining